how-to/workspace-platform-starter/client/src/framework/init-options.ts

```diff
@@ -138,15 +139,18 @@ export function registerActionListener(
 
 /**
  * Manually register a listener.
- * @param paramName The param name to look for in the init option.
  * @param handler The handler to call when a match on the param name occurs.
+ * @param paramName The param name to look for in the init option. If not specified then it is a function to be called if no params are passed.
  * @returns The subscription id that can be used to remove the listener.
  */
-export function registerListener(paramName: string, handler: InitOptionsListener): string | undefined {
+export function registerListener(handler: InitOptionsListener, paramName?: string): string | undefined {
 	if (paramName === ACTION_PARAM_NAME) {
 		logger.warn("Please use registerActionListener if you wish to listen for an action");
 		return;
 	}
+	if (isEmpty(paramName)) {
```

Comment:
Should this be !isStringValue in case someone passes ""

Reply:
done